flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/rest/SqlGatewayRestEndpointITCase.java

```diff
@@ -58,154 +58,139 @@
+                        SqlGatewayRestAPIVersion.V0);
 
-        TestResponse testResponse1 = specifiedVersionResponse.get(5, TimeUnit.SECONDS);
-        assertThat(testResponse1.getStatus()).isEqualTo("V1");
+        TestResponse testResponse1 =
+                specifiedVersionResponse.get(timeout.getSize(), timeout.getUnit());
+        assertThat(testResponse1.getStatus()).isEqualTo("V0");
 
-        // The header only support V1, send request by latest version V1
+        // The header supports V1 and V2, lets the client get the latest version as default
         CompletableFuture<TestResponse> unspecifiedVersionResponse =
                 restClient.sendRequest(
                         serverAddress.getHostName(),
                         serverAddress.getPort(),
-                        header2,
+                        header12,
                         EmptyMessageParameters.getInstance(),
                         EmptyRequestBody.getInstance(),
                         Collections.emptyList());
 
-        TestResponse testResponse2 = unspecifiedVersionResponse.get(5, TimeUnit.SECONDS);
-        assertThat(testResponse2.getStatus()).isEqualTo("V1");
+        TestResponse testResponse2 =
+                unspecifiedVersionResponse.get(timeout.getSize(), timeout.getUnit());
+        assertThat(testResponse2.getStatus()).isEqualTo("V2");
     }
 
     /** Test that requests of different version are routed to correct handlers. */
     @Test
     void testVersionSelection() throws Exception {
-        CompletableFuture<TestResponse> version1Response =
-                restClient.sendRequest(
-                        serverAddress.getHostName(),
-                        serverAddress.getPort(),
-                        header1,
-                        EmptyMessageParameters.getInstance(),
-                        EmptyRequestBody.getInstance(),
-                        Collections.emptyList(),
-                        SqlGatewayRestAPIVersion.V0);
-
-        TestResponse testResponse = version1Response.get(5, TimeUnit.SECONDS);
-        assertThat(testResponse.getStatus()).isEqualTo("V0");
-
-        CompletableFuture<TestResponse> version2Response =
-                restClient.sendRequest(
-                        serverAddress.getHostName(),
-                        serverAddress.getPort(),
-                        header2,
-                        EmptyMessageParameters.getInstance(),
-                        EmptyRequestBody.getInstance(),
-                        Collections.emptyList(),
-                        SqlGatewayRestAPIVersion.V1);
-        TestResponse testResponse2 = version2Response.get(5, TimeUnit.SECONDS);
-        assertThat(testResponse2.getStatus()).isEqualTo("V1");
+        validateVersionSelection(header0, SqlGatewayRestAPIVersion.V0);
+        validateVersionSelection(header12, SqlGatewayRestAPIVersion.V1);
+        validateVersionSelection(header12, SqlGatewayRestAPIVersion.V2);
```

Comment:
Use for-loop? In the future, we add a new version, we don't need to write another `validateVersionSelection`

Reply:
Now I think this test is very similar to the test 'testSqlGatewayMessageHeaders', because both test the message is routed correctly. So I think we can delete this test.